src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.cs

```diff
@@ -28,13 +28,24 @@ internal AnonymousTypeManager(CSharpCompilation compilation)
         /// <summary>
         /// Given anonymous type descriptor provided constructs an anonymous type symbol.
         /// </summary>
-        public NamedTypeSymbol ConstructAnonymousTypeSymbol(AnonymousTypeDescriptor typeDescr)
+        public NamedTypeSymbol ConstructAnonymousTypeSymbol(AnonymousTypeDescriptor typeDescr, BindingDiagnosticBag diagnostics)
```

Comment:
Consider renaming `diagnostics` or adding a doc comment, so it's clearer at the callsites that it's only used to collect dependencies (and hence why it's fine to pass `BindingDiagnosticBag.Discarded` sometimes)